src/hist/stack.py

```diff
@@ -30,35 +33,48 @@ def __init__(
             if first_axes != a.axes:
                 raise ValueError("The Histogram axes don't match")
 
-    def __repr__(self) -> str:
-        str_stack = ", ".join(repr(h) for h in self._stack)
-        return f"{self.__class__.__name__}({str_stack})"
+    @typing.overload
+    def __getitem__(self, val: int) -> BaseHist:
+        ...
+
+    @typing.overload
+    def __getitem__(self: T, val: slice) -> T:
+        ...
+
+    def __getitem__(self: T, val: Union[int, slice]) -> Union[BaseHist, T]:
+        if isinstance(val, slice):
+            return self.__class__(*self._stack.__getitem__(val))
 
-    def __getitem__(
-        self, val: Union[int, slice]
-    ) -> Union[BaseHist, Tuple[BaseHist, ...]]:
         return self._stack.__getitem__(val)
 
     def __iter__(self) -> Iterator[BaseHist]:
         return iter(self._stack)
 
-    def plot(self, *, overlay: None = None, **kwargs: Any) -> "List[Hist1DArtists]":
+    def __len__(self) -> int:
+        return len(self._stack)
+
+    def __repr__(self) -> str:
+        str_stack = ", ".join(repr(h) for h in self)
+        return f"{self.__class__.__name__}({str_stack})"
+
+    def plot(self, **kwargs: Any) -> "List[Hist1DArtists]":
         """
         Plot method for Stack object.
         """
-        if overlay is not None:
-            raise NotImplementedError("Currently overlay is not supported")
 
-        if self._stack[0].ndim != 1:
+        import hist.plot
```

Comment:
Besides piggy-backing on our existing mechanism for replacing this, it also makes it easy to monkeypatch in tests, too!